### PR TITLE
feat: (home page is private page)

### PR DESCRIPTION
**Pull request by nester7x**

### CODE REVIEW

File: src/screens/Registration/index.jsx [hunk 1]
```diff
@@ -91,35 +94,50 @@ const Registration = () => {
...
+          name="confirmPassword"
+          type="password"
+          inputProps={{ maxLength: 30 }}
+          value={registrationData.confirmPassword}
+          onChange={handleDataChange}
+          onBlur={(e) => blurHandler(e)}
+          placeholder="Confirm Password"
+        />
+        {isDirty.confirmPassword && (
+          <S.Error>{inputError.confirmPassword}</S.Error>
```
Comment (SerhiienkoRuslan) [inline src/screens/Registration/index.jsx]: Move validation error into Input component

File: src/components/Layout/index.jsx [hunk 2]
```diff
@@ -25,16 +25,16 @@ const Layout = ({ children }) => {
...
         </S.Link>
       ),
-      isLogin: 'general',
+      isLogin: 'loggedIn',
       key: 'home'
     },
     {
       component: (
-        <S.Link to="registration">
+        <S.Link to="">
```
Comment (SerhiienkoRuslan) [inline src/components/Layout/index.jsx]: it was better when Registration page had /registration path and Home page had "/" path

File: src/utils/LoginValidation.jsx [hunk 3]
```diff
@@ -1,10 +1,10 @@
-export const validate = (fieldName, value) => {
+export const validate = (fieldName, value, stateObj) => {
   let error = '';
   switch (fieldName) {
     case 'email':
       if (!value) {
         error = 'Email address is required';
-      } else if (!/\S+@\S+\.\S+/.test(value)) {
+      } else if (!/\S+\w*@gmail\.+com\b/.test(value)) {
```
Comment (SerhiienkoRuslan) [inline src/utils/LoginValidation.jsx]: Move this into method

isValidEmail = (value: string) => !/\S+\w*@gmail\.+com\b/.test(value)

File: src/screens/Registration/index.jsx [hunk 4]
```diff
@@ -91,35 +94,48 @@ const Registration = () => {
...
+        />
+        <S.DataInput
+          name="confirmPassword"
+          type="password"
+          inputProps={{ maxLength: 30 }}
+          value={registrationData.confirmPassword}
+          onChange={handleDataChange}
+          onBlur={(e) => blurHandler(e)}
+          placeholder="Confirm Password"
+          errorText={isDirty.confirmPassword ? inputError.confirmPassword : ''}
```
Comment (SerhiienkoRuslan) [inline src/screens/Registration/index.jsx]: isDirty.confirmPassword ? inputError.confirmPassword : ''

----

isDirty.confirmPassword && inputError.confirmPassword

**Review by aspasionov — APPROVED** [review]

File: src/components/Input/index.jsx [hunk 5]
```diff
@@ -1,10 +1,13 @@
...
 
-import InputStyled from './styles';
+import * as S from './styles';
 
-const Input = ({ onChange, value, type, ...rest }) => (
-  <InputStyled {...rest} type={type} onChange={onChange} value={value} />
+const Input = ({ onChange, value, type, errorText, ...rest }) => (
+  <>
+    <S.InputStyled {...rest} type={type} onChange={onChange} value={value} />
+    <S.Error>{errorText}</S.Error>
```
Comment (SerhiienkoRuslan) [inline src/components/Input/index.jsx]: {errorText && <S.Error>{errorText}</S.Error>}

File: src/utils/LoginValidation.jsx [hunk 6]
```diff
@@ -1,26 +1,49 @@
...
+      }
+      break;
+
+    default:
+      if (!value || !value.trim()) {
+        error.push('Username field is required');
       }
   }
-  return error;
+  return error.join('. ');
```
Comment (SerhiienkoRuslan) [inline src/utils/LoginValidation.jsx]: try .join('.\n')

**Review by SerhiienkoRuslan — APPROVED** [review]

**Pull request merged**